### PR TITLE
Add allow-non-master option to shard allocation check

### PR DESCRIPTION
**Pull request by Evesy**
#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

#### Purpose

I'd like the option to be able to run the shard allocation check on non master nodes. This means I can implement this check with a roundrobin subscription (with a proxy client) and get a meaningful result everytime.

This follows the convention of other plugins in this repository, specifically some of the metric related checks.

#### Known Compatablity Issues

None

### CODE REVIEW

**Comment by majormoses** [comment]
Thank you for your contribution! I will take a look in a sec.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachang
 
 ## [Unreleased]
 
+## [1.2.0] - 2017-03-16
```
Comment (majormoses) [inline CHANGELOG.md]: Please only add stuff under ` ## [Unreleased]` in PRs, the maintainers will decide on when to release.

File: lib/sensu-plugins-elasticsearch/version.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 module SensuPluginsElasticsearch
   module Version
     MAJOR = 1
-    MINOR = 1
```
Comment (majormoses) [inline lib/sensu-plugins-elasticsearch/version.rb]: please revert the version bump. We will do that when we are ready to release.

**Comment by Evesy** [comment]
@majormoses Thanks. I've reverted the versioning, just a conflict with other recent changelog updates that need resolving

**Comment by majormoses** [comment]
@Evesy awesome let me know when you have rebased this against master and I will be more than happy to get this over the line.

**Comment by Evesy** [comment]
@majormoses All sorted I think!

File: CHANGELOG.md~ [hunk 3]
```diff
@@ -0,0 +1,163 @@
+#Change Log
```
Comment (majormoses) [inline CHANGELOG.md~]: This is a backup file from emacs? can you remove this?

**Comment by majormoses** [comment]
@Evesy I can rebase your branch and fix it up.

**Comment by Evesy** [comment]
@majormoses Bugger, I'll get this right eventually! I've removed that now.

Thanks for bearing with me.

**Comment by majormoses** [comment]
@Evesy no problem, I welcome (sane) contributions and learning experiences

**Pull request merged**